homeassistant/components/synology_dsm/media_source.py

```diff
@@ -0,0 +1,240 @@
+            can_play=False,
+            can_expand=True,
+            children_media_class=MediaClass.DIRECTORY,
+            children=[
+                *await self._async_build_diskstations(item),
+            ],
+        )
+
+    async def _async_build_diskstations(
+        self, item: MediaSourceItem
+    ) -> list[BrowseMediaSource]:
+        """Handle browsing different diskstations."""
+        ret = []
+
+        if not item.identifier:
+            for entry in self.entries:
+                ret.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=entry.unique_id,
+                        media_class=MediaClass.DIRECTORY,
+                        media_content_type=MediaClass.IMAGE,
+                        title=f"{entry.title} - {entry.unique_id}",
+                        can_play=False,
+                        can_expand=True,
+                    )
+                )
+            return ret
+        identifier_parts = item.identifier.split("/")
+        diskstation: SynologyDSMData = self.hass.data[DOMAIN][identifier_parts[0]]
+
+        if len(identifier_parts) == 1:
+            # Get Albums
+            # The library works sync, and this expects async calls
+            try:
+                albums = await diskstation.api.photos.get_albums()
+            except SynologyDSMException:
+                return []
+
+            ret = []
+            ret.append(
+                BrowseMediaSource(
+                    domain=DOMAIN,
+                    identifier=f"{item.identifier}/0",
+                    media_class=MediaClass.DIRECTORY,
+                    media_content_type=MediaClass.IMAGE,
+                    title="All images",
+                    can_play=False,
+                    can_expand=True,
+                )
+            )
+            for album in albums:
+                ret.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=f"{item.identifier}/{album.album_id}",
+                        media_class=MediaClass.DIRECTORY,
+                        media_content_type=MediaClass.IMAGE,
+                        title=album.name,
+                        can_play=False,
+                        can_expand=True,
+                    )
+                )
+
+            return ret
+
+        # Request items of album
+        # Get Items
+        # The library works sync, and this expects async calls
+        album = SynoPhotosAlbum
+        album.album_id = int(identifier_parts[1])
+        try:
+            items = await diskstation.api.photos.get_items_from_album(album, 0, 1000)
+        except SynologyDSMException:
+            return []
+
+        ret = []
+        for items_item in items:
+            mime_type, _ = mimetypes.guess_type(items_item.file_name)
+            assert isinstance(mime_type, str)
+            parts = mime_type.split("/")
```

Comment:
`mime_type.startswith("image/")` would likely be faster and more straightforward/readable.

Reply:
I didn't know that function existed. I'm using it now